pkg/queue/manager.go

```diff
@@ -547,3 +570,36 @@ func (m *Manager) reportPendingWorkloads(cqName string, cq ClusterQueue) {
 	}
 	metrics.ReportPendingWorkloads(cqName, active, inadmissible)
 }
+
+func (m *Manager) Start(ctx context.Context) error {
+	log := ctrl.LoggerFrom(ctx).WithName("clusterQueueSnapshot")
+	ctx = ctrl.LoggerInto(ctx, log)
+
+	ticker := time.NewTicker(
+		time.Duration(m.cfg.QueueVisibility.UpdateIntervalSeconds) * time.Second,
+	)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			log.V(2).Info("Context cancelled; stop doing snapshot of cluster queue")
+			return nil
+		case t := <-ticker.C:
+			m.workloadsStatus = &kueue.ClusterQueuePendingWorkloadsStatus{
```

Comment:
This needs a lock, maybe a different lock from the one used to access the queues.